api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java

```diff
@@ -150,21 +174,31 @@ public DoubleCounter build() {
     private NoopLongCounter() {}
 
     @Override
-    public void add(long increment, String... labelKeyValuePairs) {}
+    public void add(long increment, String... labelKeyValuePairs) {
+      Utils.validateLabelPairs(labelKeyValuePairs);
+      verifyIncrement(increment);
+    }
 
     @Override
     public NoopBoundLongCounter bind(String... labelKeyValuePairs) {
       Utils.validateLabelPairs(labelKeyValuePairs);
       return NoopBoundLongCounter.INSTANCE;
     }
 
+    // Similar to add but does not verify labels.
+    private static void verifyIncrement(long increment) {
+      Utils.checkArgument(increment >= 0, "Counters can only increase");
```

Comment:
This implementation is different from the one above. You could move it up out of the inner classes and put it one level higher so it doesn't need to be repeated.

Reply:
It is different, because one uses long the other uses double.

Reply:
oh drat. stupid java.